stac_mjx/controller.py

```diff
@@ -25,11 +25,38 @@
 from tqdm import tqdm
 
 # Root position (3) + quaternion (7) in qpos
-_ROOT_QPOS_LB = -jp.inf * jp.ones(7)
-_ROOT_QPOS_UB = jp.inf * jp.ones(7)
+_ROOT_QPOS_LB = -jp.inf
+_ROOT_QPOS_UB = jp.inf
+
+# mujoco jnt_type values: https://mujoco.readthedocs.io/en/latest/APIreference/APItypes.html#mjtjoint
+_JOINT_TYPE_DIMS = {
+    0: 7,
+    1: 4,
+    2: 1,
+    3: 1,
+}
+
+
+def _align_joint_dims(types, ranges, names):
+    """Creates lower and upper bounds (as jax arrays)
+    for qpos dimensions, alongside a list of names for
+    the parts that correspond to those dimensions."""
+    lb = []
+    ub = []
+    part_names = []
+    for type, range, name in zip(types, ranges, names):
+        dims = _JOINT_TYPE_DIMS[type]
+        # Set inf bounds for freejoint
+        if type == 0:
```

Comment:
Thinking, hmm, I don't understand why we have to handle free joint separately?

Reply:
see below ⬇️ is there a better way?